programs/bpf_loader/src/helpers.rs

```diff
@@ -56,6 +80,20 @@ pub fn register_helpers<'a>(
     vm.register_helper_ex("sol_log_64", helper_sol_log_u64)?;
     vm.register_helper_ex("sol_log_64_", helper_sol_log_u64)?;
 
+    let invoke_context = Rc::new(RefCell::new(invoke_context));
+    vm.register_helper_with_context_ex(
+        "sol_process_signed_instruction_", // Rust version
+        Box::new(HelperProcessInstructionRust {
+            invoke_context: invoke_context.clone(),
+        }),
+    )?;
+    vm.register_helper_with_context_ex(
+        "sol_process_signed_instruction", // C version
+        Box::new(HelperProcessSolInstruction {
+            invoke_context: invoke_context.clone(),
+        }),
+    )?;
```

Comment:
The names here feel a little inconsistent:
* `sol_process_signed_instruction_` (rust) vs `sol_process_signed_instruction` (c).   Perhaps `sol_process_signed_instruction_rust` instead?
* `HelperProcessInstructionRust` vs  `HelperProcessSolInstruction`.  Perhaps `HelperProcessSolInstructionRust` instead?

That is, setup the C names as the default and if/when we need to hack in support for other languages, we do so with another "Language" suffix

Reply:
I think it would also really help to have a big comment block somewhere justifying why we need to treat C and Rust differently.  I found myself trying to compare the two different implementations line by line for a while, before giving up cause that's too hard.

Reply:
Agreed, a comment block would help.  It mostly boils down to the fact the C calls the function directly and rust does not and that they may need to be treated differently depending on who calls and in the case of Rust may need some translation.

For example, Rust calls `sol_process_signed_instruction` which gets translated to `sol_process_signed_instruction_` with the goal that the namespace looks the same no matter what language you are calling from.  Take a look at `sdk/src/cross_program.rs` for an example.  We also do this for all the other Rust `helpers` already.  No qualms about reworking this but probably work to do in a mother PR.

Reply:
@mvines I think we can probably clean these symbols up but doing so it outside the scope of this PR, how about a follow up one?  In this PR I'll add the Rust and C suffix to the cross-program helper names

Reply:
sure